cvmfs/quota_posix.cc

```diff
@@ -288,6 +288,10 @@ PosixQuotaManager *PosixQuotaManager::CreateShared(
   LogCvmfs(kLogQuota, kLogDebug, "trying to connect to existing pipe");
   quota_mgr->pipe_lru_[1] = open(fifo_path.c_str(), O_WRONLY | O_NONBLOCK);
   if (quota_mgr->pipe_lru_[1] >= 0) {
+
+    quota_mgr->StoreCacheMgrPid(quota_mgr->GetPid());
```

Comment:
This may be the crux of the approach: the `GetPid()` uses itself `ReadHalfPipe()`. For the fuse processes that connect to an existing cache manager, we probably want to find a different (reliable) way to get their pid.  Let's discuss.

Reply:
Yes indeed, however this already reduces the number of possibly blocking calls to ReadHalfPipe to one, and here also a (suitably long) timeout could be introduced.